### PR TITLE
Modify retry mechanism (Issue #774)

### PR DESCRIPTION
**Pull request by AngelRodriguez8008**
### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (updates to documentation, formatting, etc.)

### Description

Allow to retry commands execution also for not expected exceptions.

### Issues addressed

Allow to set remove the expensive Waiting Times, Thinking Times, Improve EasyRepro UI-Tests stability.
See detailed description in Issue #774 

### All submissions:

- [x] My code follows the code style of this project.
- [x] Do existing samples that are effected by this change still run?
- [x] I have added samples for new functionality. 
- [x] I raise detailed error messages when possible.
- [x] My code does not rely on labels that have the option to be hidden.

### Which browsers was this tested on?

- [x] Chrome
- [ ] Firefox
- [ ] IE
- [ ] Edge

### CODE REVIEW

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/ElementReference.cs [hunk 1]
```diff
@@ -184,7 +184,7 @@ public static class Elements
             { "Login_SignIn", "id(\"cred_sign_in_button\")"},
             { "Login_CrmMainPage", "//*[contains(@id,'crmTopBar') or contains(@data-id,'topBar')]"},
             { "Login_CrmUCIMainPage", "//*[contains(@data-id,'topBar')]"},
-            { "Login_StaySignedIn", "//input[@id=\"idSIButton9\"]"},
+            { "Login_StaySignedIn", "//div[@data-viewid and contains(@data-bind, 'kmsi-view')]//input[@id='idSIButton9']"},
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/DTO/ElementReference.cs:188]: Click on KMSI button, in some scenario another Dialog / Button was shown & clicked by the logic.

File: Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs [hunk 2]
```diff
@@ -5300,7 +5300,7 @@ internal void ThinkTime(TimeSpan timespan)
             ThinkTime((int)timespan.TotalMilliseconds);
         }
 
-        internal void Dispose()
+        public void Dispose()
```
Comment (AngelRodriguez8008) [inline Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs:5334]: Why internal? Browser stills open after tests is finish. (on Resharper Tests Runner)

**Comment by AngelRodriguez8008** [comment]
This is just the fist part of the improvement. After that, Waiting Times may by reduced, failing Test may take longer (because retries on the failed command). Reducing the DefaultReplyDelay may solve this problem. 

I will try to fix all UCI Sample Tests & continuously improve the performance, currently it takes 2h+ to complete a full run.

**Comment by AngelRodriguez8008** [comment]
Related to #908

**Review by TYLEROL — APPROVED** [review]

**Pull request merged**